metadata/src/main/java/org/apache/kafka/metadata/util/SnapshotFileReader.java

```diff
@@ -131,6 +132,10 @@ private void handleControlBatch(FileChannelRecordBatch batch) {
                             batch.partitionLeaderEpoch()
                         ));
                         break;
```

Comment:
Hmm. Probably outside the scope of this PR but this doesn't look correct. First, snapshot doesn't include this record. Second, KRaft mainly uses this control record so that the the current leader epoch to gets committed into the log. KRaft doesn't really read this value.

I am not sure that the `kafka-metadata-shell` should rely on the `LeaderChangeMessage` to determine the leader.